maskrcnn_benchmark/structures/bounding_box.py

```diff
@@ -232,15 +232,16 @@ def area(self):
             area = box[:, 2] * box[:, 3]
         else:
             raise RuntimeError("Should not be here")
-            
+
         return area
 
     def copy_with_fields(self, fields):
         bbox = BoxList(self.bbox, self.size, self.mode)
         if not isinstance(fields, (list, tuple)):
             fields = [fields]
         for field in fields:
-            bbox.add_field(field, self.get_field(field))
+            if self.has_field(field):
```

Comment:
Good point.

When we use `copy_with_fields` and a field is not present, should it silently ignore the field (causing an error down the line most probably?) or raise an explicit error here?

I'd lean towards raising an error here, thoughts?

Reply:
@fmassa Good catch! This check mainly aims at supporting images with missing labels (such as mask or keypoints etc.). If we raise an error here, the copy cannot be finished, and the users might need to implement another similar function by themselves.

That said, how about we add an extra argument `skip_missing=False` here? If True, it will ignore the fields if missing, otherwise by default it will raise an error.

Reply:
Does the rest of the code work if an image is missing a label? I'm curious, because I haven't taken that into account while writing the code.

If the rest of the code works fine by ignoring the label, then I'd be ok with a `skip_missing=False`. But then would this field be set to `True` in all occurrences of `copy_with_fields`?

Reply:
Currently, the method is used in computing the loss for [box_head](https://github.com/facebookresearch/maskrcnn-benchmark/blob/9cb251d340/maskrcnn_benchmark/modeling/roi_heads/box_head/loss.py#L43) and [mask_head](https://github.com/facebookresearch/maskrcnn-benchmark/blob/9cb251d340/maskrcnn_benchmark/modeling/roi_heads/mask_head/loss.py#L61). The rest of the code does not work with missing box or mask labels (if mask head is enabled). Therefore, having `skip_missing=False` by default will raise an error for these two heads. This will not break any current experiments on COCO.

When users develop new roi heads for new datasets, they can set `skip_missing=True` and handle the missing labels in the corresponding loss computation.

Reply:
Ok, this sounds good to me. Can you add the aforementioned argument, and then this will be good to go, thanks!

Reply:
Added the `skip_missing=False` argument in the revision. Thanks a lot for the discussion!